TriblerGUI/widgets/trustpage/js/drawing.js

```diff
@@ -41,45 +42,102 @@ function drawNodes(svg, data, on_click) {
         .attr("overflow", "visible")
         .attr("class", "node");
 
-    // Append a <circle> element to it.
+    // Append a <circle> element to it
     var circles = groups
         .append("circle")
         .attr("fill", function (d) {
             return getNodeColor(d, data)
         })
         .attr("r", "0")
         .attr("cx", config.node.circle.cx)
-        .attr("cy", config.node.circle.cy)
-        .style("cursor", config.node.circle.cursor)
-        .on("click", on_click)
-        .on("mouseenter", function () {
-            d3.select(this).transition().ease(d3.easeElasticOut).delay(0).duration(300).attr("r", 25);
-        }).on("mouseout", function () {
-            d3.select(this).transition().ease(d3.easeElasticOut).delay(0).duration(300).attr("r", 20);
-        });
+        .attr("cy", config.node.circle.cy);
 
     // Transition the radius of the circles
     circles.transition()
         .duration(1000)
-        .attr("r", config.node.circle.radius)
+        .attr("r", config.node.circle.radius);
 
     // Append a <text> element to it
     groups
         .append("text")
-        .attr("x", config.node.publicKeyLabel.x)
-        .attr("y", config.node.publicKeyLabel.y)
+        .attr("dominant-baseline", "central")
+        .attr("text-anchor", "middle")
         .style("font-family", config.node.publicKeyLabel.fontFamily)
         .style("font-size", config.node.publicKeyLabel.fontSize)
+        .style("font-weight", config.node.publicKeyLabel.fontWeight)
         .style("fill", config.node.publicKeyLabel.color)
         .text(function (d) {
-            return d.public_key.substr(-5);
+            return d.public_key.substr(-config.node.publicKeyLabel.characters);
+        });
+
+    // Transparent circle to capture mouse events
+    groups
+        .append("circle")
+        .style("fill-opacity", "0")
+        .attr("r", config.node.circle.radius)
+        .attr("cx", config.node.circle.cx)
+        .attr("cy", config.node.circle.cy)
+        .style("cursor", config.node.circle.cursor)
+        .on("click", on_click)
+        .on("mouseover", function (d) {
+            mouseOverNode(d, hoverInfoLabel);
+        })
+        .on("mousemove", function () {
+            hoverInfoLabel
+                .style("left", d3.event.pageX + "px")
+                .style("top", d3.event.pageY + "px");
+        })
+        .on("mouseout", function () {
+            d3.select(".hoverInfoLabel").select("table.hoverInfoTable").selectAll("tr").remove();
+            hoverInfoLabel.style("opacity", 0);
```

Comment:
Because otherwise the text label inside the node overrules the event triggering the label to appear.